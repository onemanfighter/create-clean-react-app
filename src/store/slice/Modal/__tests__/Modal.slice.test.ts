import { ModalIDs, ModalOpenState, ModalState, appStore } from '@store';
import { act, renderHook } from '@testing-library/react';

describe('Modal slice', () => {
  const onModalCloseMock = jest.fn();

  it('should return modal selector state and actions', () => {
    const modal = renderHook(() => appStore()).result.current;

    expect(modal).toMatchSnapshot();
  });

  it('should return modal state with initial value', () => {
    const { result } = renderHook(() => appStore());

    expect(result.current.Modal).toMatchSnapshot();
  });

  it('should return modal state on modal data set using openModal', () => {
    const modalData: ModalState = {
      modalID: ModalIDs.SEARCH,
      onModalClose: onModalCloseMock,
      modalOpenState: ModalOpenState.OPEN,
    };

    const { result } = renderHook(() => appStore());

    act(() => {
      result.current.Modal.openModal(modalData);
      jest.runAllTimers();
    });

    expect(result.current.Modal).toMatchSnapshot();
  });

  it.skip('should reset and close modal state on modal data set using closeModal', () => {
    const modalData: ModalState = {
      modalID: ModalIDs.SEARCH,
      onModalClose: onModalCloseMock,
      modalOpenState: ModalOpenState.OPEN,
    };
    const { result } = renderHook(() => appStore());

    act(() => {
      result.current.Modal.openModal(modalData);
      jest.runAllTimers();
    });
    expect(result.current.Modal.modalID).toEqual(modalData.modalID);
    expect(result.current.Modal.onModalClose).toEqual(modalData.onModalClose);

    act(() => {
      result.current.Modal.closeModal();
      jest.runAllTimers();
    });

    expect(result.current.Modal.modalID).toEqual(undefined);
    expect(result.current.Modal.onModalClose).toEqual(undefined);
  });
});
