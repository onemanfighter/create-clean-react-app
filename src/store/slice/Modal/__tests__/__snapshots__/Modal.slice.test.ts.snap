// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal slice should return modal selector state and actions 1`] = `
{
  "Modal": {
    "closeModal": [Function],
    "modalID": undefined,
    "modalOpenState": "CLOSE",
    "onModalClose": undefined,
    "openModal": [Function],
  },
}
`;

exports[`Modal slice should return modal state on modal data set using openModal 1`] = `
{
  "closeModal": [Function],
  "modalID": "search",
  "modalOpenState": "OPEN",
  "onModalClose": [MockFunction],
  "openModal": [Function],
}
`;

exports[`Modal slice should return modal state with initial value 1`] = `
{
  "closeModal": [Function],
  "modalID": undefined,
  "modalOpenState": "CLOSE",
  "onModalClose": undefined,
  "openModal": [Function],
}
`;
